content/departments/engineering/dev/tools/infrastructure/ci/index.md

```diff
@@ -26,9 +26,84 @@ We maintain a shared fleet of Buildkite agents for continuous integration across
 
 ### Buildkite agent queues
 
-We have several different types of agents availables. We recommend explicitly declaring which type of agent you want your jobs to run on with the `agents: { queue: "standard" }` field in your pipeline configuration.
+We have several different types of agents available. We recommend explicitly declaring which type of agent you want your jobs to run on with the `agents: { queue: "standard" }` field in your pipeline configuration.
 
 The currently available queues:
 
 - `standard`: our default Buildkite agents, currently Docker-in-Docker agents running in Kubernetes
 - `vagrant`: special Buildkite agents desgined to run resource intensive test on docker deployments.
+
+### `buildkite-job-dispatcher`
+
+> NOTE: This section serves only as an intro and overview to `buildkite-job-dispatcher`. Detailed documentation and guides should go in the [CI playbook](../../../process/incidents/playbooks/ci.md) or as close to the source code as possible.
+
+Our Buildkite agents are stateless, and are deployed in batches as Kubernetes jobs where each agent runs its workload and exits based on the size of the Buildkite backlog.
+This is managed by the `buildkite-job-dispatcher`:
+
+- [Source code](https://sourcegraph.com/github.com/sourcegraph/infrastructure/-/tree/docker-images/buildkite-job-dispatcher)
+- [Documentation, dashboards, and deployment manifests](https://github.com/sourcegraph/infrastructure/blob/main/buildkite/kubernetes/buildkite-job-dispatcher)
+
+Another potentially fragile component of this system is `buildkite-git-references`, which is a cron job and set of GCP disks that speed up pipelines by reducing the amount of cloning required.
+
+- [Source code](https://github.com/sourcegraph/infrastructure/tree/main/buildkite/buildkite-git-references)
+- [Pipeline](https://buildkite.com/sourcegraph/buildkite-git-references)
+
+Relevant runbooks:
+
+- [Replacing agents](../../../process/incidents/playbooks/ci.md#replacing-agents)
+- [Agent availability issues](../../../process/incidents/playbooks/ci.md#agent-availability-issues)
+
+A diagram overview of how the `buildkite-job-dispatcher` works (diagram adapted from [here](https://bobheadxi.dev/stateless-ci/#dynamic-kubernetes-jobs)):
+
+```mermaid
```

Comment:
couldn't get the diagram rendering :( asking for help here: https://sourcegraph.slack.com/archives/C02PSJF0QLU/p1657315582220539

Reply:
worst case I'll add a link to view this in GitHub where it renders